django/template/base.py

```diff
@@ -199,6 +199,16 @@ def compile_nodelist(self):
         except Exception as e:
             if self.engine.debug:
                 e.template_debug = self.get_exception_info(e, e.token)
+
+            if (
+                isinstance(e, TemplateSyntaxError)
+                and self.origin.name != UNKNOWN_SOURCE
```

Comment:
I can remove `and self.origin.name != UNKNOWN_SOURCE` without any test failures, can you add a test to confirm the `Template` prefix is not in error messages with an unknown source?

Reply:
oh, i did not test that 
i have changed it to try except block now test fails when `and self.origin.name != UNKNOWN_SOURCE` this condition is removed .  i think `assertRaisesMessage` checks if the string is in the message not for the equality , that is why it was passing .